File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosbo/ExtensaoBO.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 06/07/2015
' * Time: 15:02
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.IO
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Drawing.Imaging
Imports System.Windows.Forms
Imports HFSGuardaDiretorio.comum
Imports HFSGuardaDiretorio.objetos
Imports HFSGuardaDiretorio.objetosdao

Namespace objetosbo
	''' <summary>
	''' Description of ExtensaoBO.
	''' </summary>
	Public NotInheritable Class ExtensaoBO
		Private Shared m_instancia As New ExtensaoBO()

		Private diretorioBMP As Byte()
		Private arquivoBMP As Byte()

		Public Shared ReadOnly Property Instancia() As ExtensaoBO
			Get
				Return m_instancia
			End Get
		End Property

				'diretorioBMP = Rotinas.LerArquivoImagem("imagens"+Path.DirectorySeparatorChar+"diretorio.bmp");
				'arquivoBMP = Rotinas.LerArquivoImagem("imagens"+Path.DirectorySeparatorChar+"arquivo.bmp");
		Private Sub New()
		End Sub

		Public Function carregarExtensao(progressoLog As IProgressoLog) As List(Of Extensao)
			Return ExtensaoDAO.Instancia.consultarTudo(progressoLog)
		End Function

		Public Sub carregarExtensoes(lista As List(Of Extensao), listaImg16 As ImageList, listaImg32 As ImageList)
			Dim bmp16 As Bitmap, bmp32 As Bitmap

			diretorioBMP = Rotinas.imageToByteArray(listaImg16.Images(0), ImageFormat.Bmp)
			arquivoBMP = Rotinas.imageToByteArray(listaImg16.Images(2), ImageFormat.Bmp)

			For Each extensao As Extensao In lista
				bmp16 = Rotinas.byteArrayToBitmap(extensao.Bmp16)
				bmp32 = Rotinas.byteArrayToBitmap(extensao.Bmp32)

				listaImg16.Images.Add(bmp16)
				listaImg32.Images.Add(bmp32)
			Next
		End Sub

		Public Function existeExtensao(sExtensao As String, listaLocal As List(Of Extensao)) As Boolean
			If sExtensao.Trim().Length > 0 Then
				For Each extensao As Extensao In listaLocal
					If extensao.Nome.Trim().ToLower().Equals(sExtensao.Trim().ToLower()) Then
						Return True
					End If
				Next
			End If
			Return False
		End Function

		Public Function retMaxCodExtensao(listaLocal As List(Of Extensao)) As Integer
			Dim nMaior As Integer = 0
			For Each extensao As Extensao In listaLocal
				If extensao.Codigo > nMaior Then
					nMaior = extensao.Codigo
				End If
			Next
			Return nMaior + 1
		End Function

		Public Function incluirExtensao(extensao As Extensao) As Boolean
			Return (ExtensaoDAO.Instancia.incluir(extensao) > 0)
		End Function

		Public Function SalvarExtensao(listaExtensao As List(Of Extensao), sNomeDiretorio As String, sCaminhoOriginal As String, log As StringList) As Boolean
			Dim sExtensao As String
			Dim extensao As Extensao

			If Rotinas.LastDelimiter(".", sNomeDiretorio) > 0 Then
				sExtensao = Rotinas.SubString(sNomeDiretorio, Rotinas.LastDelimiter(".", sNomeDiretorio) + 1, sNomeDiretorio.Length)
				If Not existeExtensao(sExtensao, listaExtensao) Then
					extensao = New Extensao()
					extensao.Codigo = retMaxCodExtensao(listaExtensao)
					extensao.Nome = sExtensao.ToLower()
					extensao.Ordem = listaExtensao.Count + 1
					extensao.Bmp16 = Rotinas.LerArquivoImagem(sCaminhoOriginal)
					extensao.Bmp32 = Rotinas.LerArquivoImagem(sCaminhoOriginal)
					incluirExtensao(extensao)
					listaExtensao.Add(extensao)
					log.Add("Salvando Extensão: " & extensao.Nome)
					Return True
				End If
			End If
			Return False
		End Function

		Public Sub salvarExtensoes(listaDiretorio As List(Of Diretorio), listaExtensao As List(Of Extensao), progressoLog As IProgressoLog)
			Dim log As New StringList()
			Dim diretorio As Diretorio
			Dim pb As New Progresso()

			pb.Minimo = 0
			pb.Maximo = listaDiretorio.Count - 1
			pb.Posicao = 0
			pb.Passo = 1

			For i As Integer = 0 To listaDiretorio.Count - 1
				diretorio = listaDiretorio(i)

				If diretorio.Tipo.Codigo = "A"C Then
					SalvarExtensao(listaExtensao, diretorio.Nome, diretorio.CaminhoOriginal, log)
					If log.Count > 0 Then
						pb.Log = log(0)
						log.Clear()
					End If
				End If

				pb.Posicao = i

				If progressoLog IsNot Nothing Then
					progressoLog.ProgressoLog(pb)
				End If
			Next
		End Sub

		Public Function excluirExtensao(listaExtensao As List(Of Extensao), codigo As Integer) As Boolean
			Dim extensao As Extensao

			If ExtensaoDAO.Instancia.excluir(codigo) > 0 Then
				For i As Integer = 0 To listaExtensao.Count - 1
					extensao = listaExtensao(i)
					If extensao.Codigo = codigo Then
						listaExtensao.Remove(extensao)
						Exit For
					End If
				Next
				Return True
			Else
				Return False
			End If
		End Function

		Public Function excluirTodasExtensoes(listaExtensao As List(Of Extensao)) As Boolean
			If ExtensaoDAO.Instancia.excluirTudo() > 0 Then
				listaExtensao.Clear()
				Return True
			Else
				Return False
			End If
		End Function

		Public Function criarTabelaExtensoes() As Boolean
			Return (ExtensaoDAO.Instancia.criarTabela() > 0)
		End Function

		Public Function indiceExtensao(lista As List(Of Extensao), nomeExtensao As String) As Integer
			nomeExtensao = Rotinas.SubString(nomeExtensao, Rotinas.LastDelimiter(".", nomeExtensao) + 1, nomeExtensao.Length)

			For Each extensao As Extensao In lista
				If extensao.Nome.Trim().ToLower().Equals(nomeExtensao.Trim().ToLower()) Then
					Return (extensao.Ordem + 2)
				End If
			Next
			Return -1
		End Function

		Public Function pegaExtensaoPorOrdem(lista As List(Of Extensao), ordem As Integer) As Extensao
			For Each extensao As Extensao In lista
				If extensao.Ordem = ordem Then
					Return extensao
				End If
			Next
			Return Nothing
		End Function

		Public Function pegaExtensaoPorCodigo(lista As List(Of Extensao), codigo As Integer) As Extensao
			For Each extensao As Extensao In lista
				If extensao.Codigo = codigo Then
					Return extensao
				End If
			Next
			Return Nothing
		End Function

		Public Function pegaExtensaoPorNome(lista As List(Of Extensao), nome As String, tipo As String) As Extensao
			Dim ext As Extensao = Nothing

			If tipo.Equals("Diretório") Then
				ext = New Extensao()
				ext.Bmp16 = diretorioBMP
			Else
				If lista IsNot Nothing AndAlso lista.Count > 0 Then
					For Each extensao As Extensao In lista
						If extensao.Nome.ToLower().Equals(nome.ToLower()) Then
							Return extensao
						End If
					Next
				End If
				If tipo.Equals("Arquivo") Then
					ext = New Extensao()
					ext.Bmp16 = arquivoBMP
				End If
			End If
			Return ext
		End Function

		Public Function ExportarExtensao(tipo As TipoExportarExtensao, listaExtensao As List(Of Extensao)) As Boolean
			Dim sCaminho As String, sImg16 As String, sImg32 As String, sExtensao As String

			If listaExtensao.Count > 0 Then
				sCaminho = Rotinas.ExtractFilePath(Application.ExecutablePath) & Rotinas.BARRA_INVERTIDA

				If tipo = TipoExportarExtensao.teBMP Then
					sExtensao = "." & Rotinas.EXTENSAO_BMP
				ElseIf tipo = TipoExportarExtensao.teICO Then
					sExtensao = "." & Rotinas.EXTENSAO_ICO
				ElseIf tipo = TipoExportarExtensao.teGIF Then
					sExtensao = "." & Rotinas.EXTENSAO_GIF
				ElseIf tipo = TipoExportarExtensao.teJPG Then
					sExtensao = "." & Rotinas.EXTENSAO_JPEG
				ElseIf tipo = TipoExportarExtensao.tePNG Then
					sExtensao = "." & Rotinas.EXTENSAO_PNG
				ElseIf tipo = TipoExportarExtensao.teTIF Then
					sExtensao = "." & Rotinas.EXTENSAO_TIFF
				Else
					sExtensao = ".img"
				End If

				For Each extensao As Extensao In listaExtensao
					sImg16 = sCaminho & extensao.Nome & "16" & sExtensao
					If Rotinas.FileExists(sImg16) Then
						Rotinas.DeleteFile(sImg16)
					End If

					sImg32 = sCaminho & extensao.Nome & "32" & sExtensao
					If Rotinas.FileExists(sImg32) Then
						Rotinas.DeleteFile(sImg32)
					End If

					Select Case tipo
						Case TipoExportarExtensao.teBMP
							If True Then
								Rotinas.SaveToFile(extensao.Bmp16, sImg16)
								Rotinas.SaveToFile(extensao.Bmp32, sImg32)
							End If
							Exit Select
						Case TipoExportarExtensao.teICO
							If True Then
								If extensao.Bmp16 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp16, Rotinas.EXTENSAO_ICO), sImg16)
								End If
								If extensao.Bmp32 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp32, Rotinas.EXTENSAO_ICO), sImg32)
								End If
							End If
							Exit Select
						Case TipoExportarExtensao.teGIF
							If True Then
								If extensao.Bmp16 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp16, Rotinas.EXTENSAO_GIF), sImg16)
								End If
								If extensao.Bmp32 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp32, Rotinas.EXTENSAO_GIF), sImg32)
								End If
							End If
							Exit Select
						Case TipoExportarExtensao.teJPG
							If True Then
								If extensao.Bmp16 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp16, Rotinas.EXTENSAO_JPEG), sImg16)
								End If
								If extensao.Bmp32 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp32, Rotinas.EXTENSAO_JPEG), sImg32)
								End If
							End If
							Exit Select
						Case TipoExportarExtensao.tePNG
							If True Then
								If extensao.Bmp16 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp16, Rotinas.EXTENSAO_PNG), sImg16)
								End If
								If extensao.Bmp32 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp32, Rotinas.EXTENSAO_PNG), sImg32)
								End If
							End If
							Exit Select
						Case TipoExportarExtensao.teTIF
							If True Then
								If extensao.Bmp16 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp16, Rotinas.EXTENSAO_TIFF), sImg16)
								End If
								If extensao.Bmp32 IsNot Nothing Then
									Rotinas.SaveToFile(Rotinas.BmpParaImagem(extensao.Bmp32, Rotinas.EXTENSAO_TIFF), sImg32)
								End If
							End If
							Exit Select

					End Select
				Next
				Return True
			End If
			Return False
		End Function

		Public Sub ImportarExtensao(caminho As String, listaExtensao As List(Of Extensao))
			Dim listaArquivos As StringList, log As StringList
			listaArquivos = New StringList()
			log = New StringList()

			If DiretorioBO.Instancia.carregarArquivos(caminho, listaArquivos) Then
				For Each sArquivo As String In listaArquivos
					SalvarExtensao(listaExtensao, Rotinas.ExtractFileName(sArquivo), sArquivo, log)
				Next
			End If

		End Sub

		Public Sub ExtrairExtensao(caminho As String, listaExtensao As List(Of Extensao))
			Dim listaArquivos As StringList, log As StringList
			listaArquivos = New StringList()
			log = New StringList()

			If DiretorioBO.Instancia.carregarTodosArquivos(caminho, listaArquivos) Then
				For Each sArquivo As String In listaArquivos
					SalvarExtensao(listaExtensao, Rotinas.ExtractFileName(sArquivo), sArquivo, log)
				Next
			End If
		End Sub

	End Class
End Namespace

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosbo/ExtensaoBO.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 06/07/2015
 * Time: 15:02
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.IO;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Imaging;
using System.Windows.Forms;
using HFSGuardaDiretorio.comum;
using HFSGuardaDiretorio.objetos;
using HFSGuardaDiretorio.objetosdao;

namespace HFSGuardaDiretorio.objetosbo
{
	/// <summary>
	/// Description of ExtensaoBO.
	/// </summary>
	public sealed class ExtensaoBO
	{
		private static ExtensaoBO instancia = new ExtensaoBO();
		
		private byte[] diretorioBMP;
    	private byte[] arquivoBMP;
		
		public static ExtensaoBO Instancia {
			get {
				return instancia;
			}
		}
		
		private ExtensaoBO()
		{
			//diretorioBMP = Rotinas.LerArquivoImagem("imagens"+Path.DirectorySeparatorChar+"diretorio.bmp");
    	    //arquivoBMP = Rotinas.LerArquivoImagem("imagens"+Path.DirectorySeparatorChar+"arquivo.bmp");
		}
		
		public List<Extensao> carregarExtensao(IProgressoLog progressoLog) {
			return ExtensaoDAO.Instancia.consultarTudo(progressoLog);
		}
		
		public void carregarExtensoes(List<Extensao> lista, 
			ImageList listaImg16, ImageList listaImg32){
			Bitmap bmp16, bmp32;
			
			diretorioBMP = Rotinas.imageToByteArray(listaImg16.Images[0], ImageFormat.Bmp);
			arquivoBMP = Rotinas.imageToByteArray(listaImg16.Images[2], ImageFormat.Bmp);
			
			foreach (Extensao extensao in lista) {
				bmp16 = Rotinas.byteArrayToBitmap(extensao.Bmp16);
				bmp32 = Rotinas.byteArrayToBitmap(extensao.Bmp32);
				
				listaImg16.Images.Add(bmp16);
				listaImg32.Images.Add(bmp32);
			}
		}

		public bool existeExtensao(string sExtensao,
				List<Extensao> listaLocal) {
			if (sExtensao.Trim().Length > 0) {
				foreach (Extensao extensao in listaLocal) {
					if (extensao.Nome.Trim().ToLower().Equals(
							sExtensao.Trim().ToLower())) {
						return true;
					}
				}
			}
			return false;
		}

		public int retMaxCodExtensao(List<Extensao> listaLocal) {
			int nMaior = 0;
			foreach (Extensao extensao in listaLocal) {
				if (extensao.Codigo > nMaior) {
					nMaior = extensao.Codigo;
				}
			}
			return nMaior + 1;
		}

		public bool incluirExtensao(Extensao extensao)  {
			return (ExtensaoDAO.Instancia.incluir(extensao) > 0);
		}

		public bool SalvarExtensao(List<Extensao> listaExtensao,
				string sNomeDiretorio, string sCaminhoOriginal, StringList log)
		{
			string sExtensao;
			Extensao extensao;

			if (Rotinas.LastDelimiter(".", sNomeDiretorio) > 0) {
				sExtensao = Rotinas.SubString(sNomeDiretorio,
						Rotinas.LastDelimiter(".", sNomeDiretorio) + 1,
						sNomeDiretorio.Length);
				if (!existeExtensao(sExtensao, listaExtensao)) {
					extensao = new Extensao();
					extensao.Codigo = retMaxCodExtensao(listaExtensao);
					extensao.Nome = sExtensao.ToLower();
					extensao.Ordem = listaExtensao.Count+1;
					extensao.Bmp16 = Rotinas.LerArquivoImagem(sCaminhoOriginal);
					extensao.Bmp32 = Rotinas.LerArquivoImagem(sCaminhoOriginal);
					incluirExtensao(extensao);                
					listaExtensao.Add(extensao);
					log.Add("Salvando Extensão: " + extensao.Nome);
					return true;
				}
			}
			return false;
		}

		public void salvarExtensoes(List<Diretorio> listaDiretorio,
				List<Extensao> listaExtensao, 
				IProgressoLog progressoLog)  {
			StringList log = new StringList();
			Diretorio diretorio;
			Progresso pb = new Progresso();

			pb.Minimo = 0;
			pb.Maximo = listaDiretorio.Count - 1;
			pb.Posicao = 0;
			pb.Passo = 1;

			for (int i = 0; i < listaDiretorio.Count; i++) {
				diretorio = listaDiretorio[i];

				if (diretorio.Tipo.Codigo == 'A') {
					SalvarExtensao(listaExtensao, diretorio.Nome,
							diretorio.CaminhoOriginal, log);
					if (log.Count > 0){
						pb.Log = log[0];
						log.Clear();
					}
				}

				pb.Posicao = i;

				if (progressoLog != null) {
					progressoLog.ProgressoLog(pb);
				}
			}
		}

		public bool excluirExtensao(List<Extensao> listaExtensao,
				int codigo)  {
			Extensao extensao;

			if (ExtensaoDAO.Instancia.excluir(codigo) > 0) {
				for (int i = 0; i < listaExtensao.Count; i++) {
					extensao = listaExtensao[i];
					if (extensao.Codigo == codigo) {
						listaExtensao.Remove(extensao);
						break;
					}
				}
				return true;
			} else {
				return false;
			}
		}

		public bool excluirTodasExtensoes(List<Extensao> listaExtensao)
				 {
			if (ExtensaoDAO.Instancia.excluirTudo() > 0) {
				listaExtensao.Clear();
				return true;
			} else {
				return false;
			}
		}

		public bool criarTabelaExtensoes()  {
			return (ExtensaoDAO.Instancia.criarTabela() > 0);
		}

		public int indiceExtensao(List<Extensao> lista,
				string nomeExtensao) {
			nomeExtensao = Rotinas.SubString(nomeExtensao,
					Rotinas.LastDelimiter(".", nomeExtensao) + 1,
					nomeExtensao.Length);

			foreach (Extensao extensao in lista) {
				if (extensao.Nome.Trim().ToLower().Equals(
						nomeExtensao.Trim().ToLower())) {
					return (extensao.Ordem + 2);
				}
			}
			return -1;
		}

		public Extensao pegaExtensaoPorOrdem(List<Extensao> lista,
				int ordem) {
			foreach (Extensao extensao in lista) {
				if (extensao.Ordem == ordem) {
					return extensao;
				}
			}
			return null;
		}

		public Extensao pegaExtensaoPorCodigo(List<Extensao> lista,
				int codigo) {
			foreach (Extensao extensao in lista) {
				if (extensao.Codigo == codigo) {
					return extensao;
				}
			}
			return null;
		}

		public Extensao pegaExtensaoPorNome(List<Extensao> lista,
				string nome, string tipo) {
			Extensao ext = null;
			
			if (tipo.Equals("Diretório")){
				ext = new Extensao();
				ext.Bmp16 = diretorioBMP;
			} else {        
				if (lista != null && lista.Count > 0) {
					foreach (Extensao extensao in lista) {
						if (extensao.Nome.ToLower().Equals(
								nome.ToLower())) {
							return extensao;
						}
					}
				}
				if (tipo.Equals("Arquivo")){
					ext = new Extensao();
					ext.Bmp16 = arquivoBMP;
				}            
			}
			return ext;
		}
		
		public bool ExportarExtensao(TipoExportarExtensao tipo,
				List<Extensao> listaExtensao) {
			string sCaminho, sImg16, sImg32, sExtensao;
			
			if (listaExtensao.Count > 0) {
				sCaminho = Rotinas.ExtractFilePath(Application.ExecutablePath) +
						Rotinas.BARRA_INVERTIDA;

				if (tipo == TipoExportarExtensao.teBMP) {
					sExtensao = "." + Rotinas.EXTENSAO_BMP;
				} else if (tipo == TipoExportarExtensao.teICO) {
					sExtensao = "." + Rotinas.EXTENSAO_ICO;
				} else if (tipo == TipoExportarExtensao.teGIF) {
					sExtensao = "." + Rotinas.EXTENSAO_GIF;
				} else if (tipo == TipoExportarExtensao.teJPG) {
					sExtensao = "." + Rotinas.EXTENSAO_JPEG;
				} else if (tipo == TipoExportarExtensao.tePNG) {
					sExtensao = "." + Rotinas.EXTENSAO_PNG;
				} else if (tipo == TipoExportarExtensao.teTIF) {
					sExtensao = "." + Rotinas.EXTENSAO_TIFF;
				} else {
					sExtensao = ".img";
				}

				foreach (Extensao extensao in listaExtensao) {
					sImg16 = sCaminho + extensao.Nome + "16" + sExtensao;
					if (Rotinas.FileExists(sImg16)) {
						Rotinas.DeleteFile(sImg16);
					}

					sImg32 = sCaminho + extensao.Nome + "32" + sExtensao;
					if (Rotinas.FileExists(sImg32)) {
						Rotinas.DeleteFile(sImg32);
					}

					switch (tipo) {
						case TipoExportarExtensao.teBMP: {
							Rotinas.SaveToFile(extensao.Bmp16, sImg16);
							Rotinas.SaveToFile(extensao.Bmp32, sImg32);
						}
						break;
						case TipoExportarExtensao.teICO: {
							if (extensao.Bmp16 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp16,
										Rotinas.EXTENSAO_ICO), sImg16);
							}
							if (extensao.Bmp32 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp32,
										Rotinas.EXTENSAO_ICO), sImg32);
							}
						}
						break;
						case TipoExportarExtensao.teGIF: {
							if (extensao.Bmp16 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp16,
										Rotinas.EXTENSAO_GIF), sImg16);
							}
							if (extensao.Bmp32 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp32,
										Rotinas.EXTENSAO_GIF), sImg32);
							}
						}
						break;
						case TipoExportarExtensao.teJPG: {
							if (extensao.Bmp16 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp16,
										Rotinas.EXTENSAO_JPEG), sImg16);
							}
							if (extensao.Bmp32 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp32,
										Rotinas.EXTENSAO_JPEG), sImg32);
							}
						}
						break;
						case TipoExportarExtensao.tePNG: {
							if (extensao.Bmp16 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp16,
										Rotinas.EXTENSAO_PNG), sImg16);
							}
							if (extensao.Bmp32 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp32,
										Rotinas.EXTENSAO_PNG), sImg32);
							}
						}
						break;
						case TipoExportarExtensao.teTIF: {
							if (extensao.Bmp16 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp16,
										Rotinas.EXTENSAO_TIFF), sImg16);
							}
							if (extensao.Bmp32 != null) {
								Rotinas.SaveToFile(
										Rotinas.BmpParaImagem(extensao.Bmp32,
										Rotinas.EXTENSAO_TIFF), sImg32);
							}
						}
						break;
					}

				}
				return true;
			}
			return false;
		}

		public void ImportarExtensao(string caminho,
				List<Extensao> listaExtensao) {
			StringList listaArquivos, log;
			listaArquivos = new StringList();
			log = new StringList();
			
			if (DiretorioBO.Instancia.carregarArquivos(caminho, listaArquivos)) {
				foreach (string sArquivo in listaArquivos) {
					SalvarExtensao(listaExtensao, Rotinas.ExtractFileName(sArquivo),
							sArquivo, log);
				}
			}

		}

		public void ExtrairExtensao(string caminho,
				List<Extensao> listaExtensao)  {
			StringList listaArquivos, log;
			listaArquivos = new StringList();
			log = new StringList();

			if (DiretorioBO.Instancia.carregarTodosArquivos(caminho, listaArquivos)) {
				foreach (string sArquivo in listaArquivos) {
					SalvarExtensao(listaExtensao, Rotinas.ExtractFileName(sArquivo),
							sArquivo, log);
				}
			}
		}
		
	}
}
